tests/test_config.py

```diff
@@ -21,19 +22,19 @@
 
 
 def _reload_config() -> None:
-    skypilot_config._dict = None
+    skypilot_config._dict = skypilot_config.NestedConfig()
+    skypilot_config._loaded_config_path = None
     skypilot_config._try_load_config()
 
 
 def _check_empty_config() -> None:
     """Check that the config is empty."""
-    assert not skypilot_config.loaded()
+    assert not skypilot_config.loaded(), (skypilot_config._dict,
+                                          skypilot_config._loaded_config_path)
     assert skypilot_config.get_nested(
         ('aws', 'ssh_proxy_command'), None) is None
     assert skypilot_config.get_nested(('aws', 'ssh_proxy_command'),
                                       'default') == 'default'
-    with pytest.raises(RuntimeError):
-        skypilot_config.set_nested(('aws', 'ssh_proxy_command'), 'value')
```

Comment:
Hmm, set_nested()'s prev and curr impl both say it doesn't raise an error. Why did this test pass?

Reply:
This is due to the change for `_check_loaded_or_die`. Previously, if a user does not have `~/.sky/config.yaml`, we will raise if `set_nested` is called, but it doesn't make sense if we really want to change the config from empty.